source.go

```diff
@@ -134,81 +157,36 @@ func (s *Source) nextPage(ctx context.Context) (sdk.Record, error) {
 		Msg("fetching page")
 
 	// fetch the page and then all of its children
-	page, err := s.client.Page.Get(ctx, notion.PageID(id))
-	if err != nil {
-		// The search endpoint that we use to list all the pages
-		// can return stale results.
-		// It's also possible that a page has been deleted after
-		// we got the ID but before we actually read the whole page.
-		if s.notFound(err) {
-			sdk.Logger(ctx).Info().
-				Str("block_id", id).
-				Msg("the resource does not exist or the resource has not been shared with owner of the token")
-
-			return s.nextPage(ctx)
-		}
-
-		return sdk.Record{}, fmt.Errorf("failed fetching page %v: %w", id, err)
+	pg, err := s.client.GetPage(ctx, id)
+	// The search endpoint that we use to list all the pages
+	// can return stale results.
+	// It's also possible that a page has been deleted after
+	// we got the ID but before we actually read the whole page.
+	if errors.Is(err, client.ErrPageNotFound) {
+		sdk.Logger(ctx).Info().
```

Comment:
Shouldn't this be a warning?

Reply:
The 2 possibilities mentioned in the log message are:
1. the page was deleted (but not yet removed from the search index) 
2. or the access to the page was removed. 

The second is probably not so probable, because it means we found the page via search, and access to it was removed after we found it via search and before we tried fetching it. So most of 404s I believe will be because the page was deleted but left in search index, which may not be for a warning, but I don't have a strong opinion about that.